code/frameworks/nextjs/src/resolved-router-context.ts

```diff
@@ -0,0 +1,12 @@
+// this file will be aliased by webpack at runtime (this is just for typing)
+
+import { RouterContext as RouterContextV12 } from 'next/dist/shared/lib/router-context';
+import { RouterContext as RouterContextV11 } from 'nextv11/dist/next-server/lib/router-context';
```

Comment:
I'd recommend taking out the extra types. Including the past versions of nextjs fires all kinds of alarms for security scans even though you only use it for type checking. Integration tests, I've found, are perfectly fine to ensure proper usage of the router.

Reply:
An attempt at this is here: https://github.com/storybookjs/storybook/commit/af0477dcc4602a9d38435d11568f5981cf8b3012